lib/hockey_gerrit.rb

```diff
@@ -1,103 +1,104 @@
 
-    def git_commit_sha
-      sha = `git rev-parse --verify HEAD`
-      raise 'command failed' unless $CHILD_STATUS.success?
-      sha
-    end
+  def gerrit_refspec
+    raise 'Must set GERRIT_REFSPEC' unless gerrit_env && !gerrit_env.empty?
+    gerrit_split = gerrit_env.split('/')
+    raise 'GERRIT_REFSPEC is invalid' unless gerrit_split.size >= 2
+    change, patch = gerrit_split[-2..-1]
+    "g#{change},#{patch}"
+  end
 
-    HOCKEY_TOKEN = ENV['HOCKEY_TOKEN']
-    IPA = ARGV.first
-    BUILD_URL = ENV['BUILD_URL']
-
-    def configure_options
-      gerrit = gerrit_refspec
-      log = git_log
-      release_notes = "#{gerrit}\n#{log}"
-
-      markdown = 1
-      available_for_download = 2
-      dont_notify = 0
-      options = {
-          notes_type: markdown,
-          notes: release_notes,
-          notify: dont_notify,
-          ipa: IPA,
-          status: available_for_download,
-          build_server_url: BUILD_URL,
-          commit_sha: git_commit_sha
-      }
-
-      dysm = IPA.to_s.gsub('ipa', 'app.dSYM.zip')
-      if File.exist?(dysm)
-        options[:dsym] = dysm
-      else
-        is_android = File.extname(IPA) == '.apk'
-        puts 'dSYM not found! Unable to symbolicate crashes' unless is_android
-      end
-      options
-    end
+  def git_log
+    git_log = `git log --reverse -1 --format="%an: %s"`
+    raise 'command failed' unless $CHILD_STATUS.success?
+    git_log
+  end
 
-    def hockey_url(response)
-      config_url = response.body ? response.body['config_url'] : ''
-      raise 'Missing config_url' unless config_url
-      config_url.gsub!('https://upload.hockeyapp.net/', 'https://rink.hockeyapp.net/')
-      config_url
-    end
+  def git_commit_sha
+    sha = `git rev-parse --verify HEAD`
+    raise 'command failed' unless $CHILD_STATUS.success?
+    sha
+  end
 
-    def validate_args
-      raise 'Must set HOCKEY_TOKEN' unless HOCKEY_TOKEN
-      raise 'Must provide path to IPA' unless IPA
-      raise 'IPA doesn\'t exist' unless File.exist?(IPA)
-      raise 'Must provide BUILD_URL' unless BUILD_URL
+  def configure_options
+    gerrit = gerrit_refspec
+    log = git_log
+    release_notes = "#{gerrit}\n\n#{log}"
+
+    markdown = 1
+    available_for_download = 2
+    dont_notify = 0
+    options = {
+        notes_type: markdown,
+        notes: release_notes,
+        notify: dont_notify,
+        ipa: ipa,
+        status: available_for_download,
+        build_server_url: build_url,
+        commit_sha: git_commit_sha
+    }
+
+    dsym_ext = 'app.dSYM.zip'
+    dsym = ipa.to_s.gsub('ipa', dsym_ext)
+
+    if File.exist?(dsym) && dsym.end_with?(dsym_ext)
+      options[:dsym_filename] = dsym
+    else
+      is_android = File.extname(ipa) == '.apk'
+      puts 'dSYM not found! Unable to symbolicate crashes' unless is_android
     end
+    options
+  end
+
+  def hockey_url(response)
+    config_url = response.body ? response.body['config_url'] : ''
```

Comment:
If response.body is falsey and config_url is set to an empty string, won't that just cause other things to break down the line?